modules/brew/brew.sh

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Convince the installer that we are in CI
+touch /.dockerenv
+
+# Debugging
+DEBUG="${DEBUG:-false}"
+if [[ "${DEBUG}" == true ]]; then
+    set -x
+fi
+
+# Check if gcc is installed
+if ! command -v gcc &> /dev/null
+then
+    echo "ERROR: \"gcc\" package could not be found."
+    echo "       Brew depends on \"gcc\" in order to function"
+    echo "       Please include \"gcc\" in the list of packages to install with the system package manager"
+    exit 1
+fi
+
+# Module-specific directories and paths
+MODULE_DIRECTORY="${MODULE_DIRECTORY:-/tmp/modules}"
+
+# Configuration values
+UPDATE_INTERVAL=$(echo "${1}" | yq -I=0 ".update-interval")
+if [[ -z "${UPDATE_INTERVAL}" || "${UPDATE_INTERVAL}" == "null" ]]; then
+    UPDATE_INTERVAL="6h"
+fi
+
+UPGRADE_INTERVAL=$(echo "$1" | yq -I=0 ".upgrade-interval")
+if [[ -z "${UPGRADE_INTERVAL}" || "${UPGRADE_INTERVAL}" == "null" ]]; then
+    UPGRADE_INTERVAL="8h"
+fi
+
+WAIT_AFTER_BOOT_UPDATE=$(echo "${1}" | yq -I=0 ".wait-after-boot-update")
+if [[ -z "${WAIT_AFTER_BOOT_UPDATE}" || "${WAIT_AFTER_BOOT_UPDATE}" == "null" ]]; then
+    WAIT_AFTER_BOOT_UPDATE="10min"
+fi
+
+WAIT_AFTER_BOOT_UPGRADE=$(echo "${1}" | yq -I=0 ".wait-after-boot-upgrade")
+if [[ -z "${WAIT_AFTER_BOOT_UPGRADE}" || "${WAIT_AFTER_BOOT_UPGRADE}" == "null" ]]; then
+    WAIT_AFTER_BOOT_UPGRADE="30min"
+fi
+
+AUTO_UPDATE=$(echo "${1}" | yq -I=0 ".auto-update")
+if [[ -z "${AUTO_UPDATE}" || "${AUTO_UPDATE}" == "null" ]]; then
+    AUTO_UPDATE=true
+fi
+
+AUTO_UPGRADE=$(echo "${1}" | yq -I=0 ".auto-upgrade")
+if [[ -z "${AUTO_UPGRADE}" || "${AUTO_UPGRADE}" == "null" ]]; then
+    AUTO_UPGRADE=true
+fi
+
+NOFILE_LIMITS=$(echo "${1}" | yq -I=0 ".nofile-limits")
+if [[ -z "${NOFILE_LIMITS}" || "${NOFILE_LIMITS}" == "null" ]]; then
+    NOFILE_LIMITS=false
+fi
+
+BREW_ANALYTICS=$(echo "${1}" | yq -I=0 ".brew-analytics")
+if [[ -z "${BREW_ANALYTICS}" || "${BREW_ANALYTICS}" == "null" ]]; then
+    BREW_ANALYTICS=true
+fi
+
+# Create necessary directories
+mkdir -p /var/home
+mkdir -p /var/roothome
+
+# Always install Brew
+echo "Downloading and installing Brew..."
+curl -Lo /tmp/brew-install https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh
```

Comment:
About verbose logging,

I analyzed the install script & I don't see anything that can solve this.

Might as well live with it.

What y'all think?

Reply:
Great work. 

Once you can confirm that the script works with all the existing flags, @xynydev or @gmpinder can merge this PR.

Reply:
Yeah, I will test other things too.